e2e/kubectl.e2e.spec.ts

```diff
@@ -0,0 +1,117 @@
+
+let page: Page;
+const defaultReportFolder = path.join(__dirname, 'reports/');
+
+test.describe.serial('Rancher Desktop - K8s Deploy Test', () => {
+  let mainTitle: Locator;
+  let utils: TestUtils;
+  let electronApp: ElectronApplication;
+  let context: BrowserContext;
+
+  const mainTitleSelector = '[data-test="mainTitle"]';
+
+  test.beforeAll(async() => {
+    utils = new TestUtils();
+    utils.createDefaultSettings();
+
+    electronApp = await _electron.launch({
+      args: [
+        path.join(__dirname, '../'),
+        '--disable-gpu',
+        '--whitelisted-ips=',
+        '--disable-dev-shm-usage',
+      ]
+    });
+    context = electronApp.context();
+
+    await context.tracing.start({ screenshots: true, snapshots: true });
+    page = await electronApp.firstWindow();
+  });
+
+  test.afterAll(async() => {
+    await context.tracing.stop({ path: `${ defaultReportFolder }pw-trace.zip` });
+    await electronApp.close();
+  });
+
+  test('should load Rancher Desktop App', async() => {
+    mainTitle = page.locator(mainTitleSelector);
+
+    await expect(mainTitle).toHaveText('Welcome to Rancher Desktop');
+  });
+
+  test('should start loading the background services', async() => {
+    const progressBarSelector = page.locator('.progress');
+
+    await progressBarSelector.waitFor({ state: 'detached', timeout: 300_000 });
+    await expect(progressBarSelector).toBeHidden();
+  });
+
+  test('should run Kubernetes on Rancher Desktop (kubectl)', async() => {
+    // CI take longer than local env to start up
+    // giving it some time to proper start the background services
+    if (process.env.CI) {
+      console.log('Waiting for services - CI take a while to load...');
+      await utils.delay(200_000);
+    }
+    const output = await tools.kubectl('cluster-info');
+
+    expect(output).toMatch(/is running at ./);
+  });
+
+  test('should create a sample namespace', async() => {
+    const getExistingNamespaces = (await tools.kubectl('get', 'namespace', '--output=name')).trim();
+
+    try {
+      if (getExistingNamespaces.includes('rd-nginx-demo')) {
+        await tools.kubectl('delete', 'ns', 'rd-nginx-demo');
+      }
+      await tools.kubectl('create', 'namespace', 'rd-nginx-demo');
+    } finally {
+      const namespaces = (await tools.kubectl('get', 'namespace', '--output=name')).trim();
+
+      expect(namespaces).toMatch(/rd-nginx-demo/);
+    }
+  });
+  test('should deploy sample nginx server', async() => {
+    try {
+      const yamlFilePath = path.join(path.dirname(__dirname), 'e2e', 'assets', 'k8s-deploy-sample', 'nginx-sample-app.yaml');
+
+      await tools.kubectl('apply', '-f', yamlFilePath, '-n', 'rd-nginx-demo');
+
+      for (let i = 0; i < 10; i++) {
+        const podName = (await tools.kubectl('get', 'pods', '--output=name', '-n', 'rd-nginx-demo')).trim();
+
+        if (podName) {
+          expect(podName).not.toBeFalsy();
+          break;
+        }
+        await utils.delay(5_000);
```

Comment:
Maybe it would have made sense to run `kubectl apply --wait`?

Reply:
Actually it won't work.
Here's why:
Even passing `--wait`, the pod will take a few seconds to shows up. It roughly takes ~1.2sec to be available. Things happens too fast and it will fail on the next assertion. Didn't find a better solution to rely on kubectl to wait until the pod being ready.

Also the wait statement seems strange for me:
[kubectl apply wait](https://kubernetes.io/docs/reference/generated/kubectl/kubectl-commands#apply)
```bash
wait |   | false | If true, wait for resources to be gone before returning. This waits for finalizers.
```
Seems that will wait for the resource be gone and then it will start again... sounds confusing...

Reply:
Ah, okay, so the deployment is done but it's not ready.
In which case `kubectl wait --for=condition=Ready …` might be useful (as long as the resource you're waiting for exists).  Might need to wait for the deployment itself instead of the pod.